src/nvim/api/extmark.c

```diff
@@ -705,6 +710,13 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id, Integer line, Integer
   bool ephemeral = false;
   OPTION_TO_BOOL(ephemeral, ephemeral, false);
 
+  OPTION_TO_BOOL(decor.ui_watched, ui_watched, false);
+  if (decor.ui_watched && !kv_size(decor.virt_text)) {
```

Comment:
we don't need this. Instead, in `draw_virt_text` change `kv_size(item->decor.virt_text)` to `kv_size(item->decor.virt_text) || item->decor.ui_watched`

Reply:
Tried that, didn't work. The mark is filtered earlier.

Reply:
likely. there are a bunch of places in decoration.c which check `kv_size(decor.virt_text)`. I think this code can be simplified actually as we now use `marktree_decor_level` to indicate if a extmark is decorative or not (so these checks can just be removed)

Reply:
Let me first try if I can glue something to all these kv_sizes...

Reply:
Oka it works. I'm adding a new commit (to be squashed later)